File: frontend/src/models/item/item-hold.ts
// Copyright (c) Microsoft Corporation.
// Licensed under the MIT license.

import { ItemHoldDTO } from '../../data-services/api-services/models/item-hold-dto';

export class ItemHold {
    held: string = '';

    ownerId: string = '';

    queueId: string = '';

    queueViewId: string = '';

    fromDTO(itemHold: ItemHoldDTO) {
        const {
            held,
            ownerId,
            queueId,
            queueViewId
        } = itemHold;

        this.held = held;
        this.ownerId = ownerId;
        this.queueId = queueId;
        this.queueViewId = queueViewId;

        return this;
    }
}

File: backend/model/src/main/java/com/griddynamics/msd365fp/manualreview/model/ItemHold.java
// Copyright (c) Microsoft Corporation.
// Licensed under the MIT license.

package com.griddynamics.msd365fp.manualreview.model;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.io.Serializable;
import java.time.OffsetDateTime;

@AllArgsConstructor
@NoArgsConstructor
@Builder
@Data
public class ItemHold implements Serializable  {
    private OffsetDateTime held;
    private String ownerId;
    private String queueId;
    private String queueViewId;
}
